package/MDAnalysis/lib/distances.py

```diff
@@ -1549,3 +1551,45 @@ def apply_PBC(coords, box, backend="serial"):
         _run("triclinic_pbc", args=(coords, box), backend=backend)
 
     return coords
+
+
+@check_coords('vectors', enforce_copy=False, enforce_dtype=False)
+def minimize_vectors(vectors, box):
+    """Apply minimum image convention to an array of vectors
+
+    This function is required for calculating the correct vectors between two
+    points.  A naive approach of ``ag1.positions - ag2.positions`` will not
+    provide the minimum vectors between particles, even if all particles are
+    within the primary unit cell (box).
+
+    Parameters
+    ----------
+    vectors : numpy.ndarray
+        Vector array of shape ``(n, 3)``, either float32 or float64.  These
+        represent many vectors (such as between two particles).
+    box : numpy.ndarray
+        The unitcell dimensions of the system, which can be orthogonal or
+        triclinic and must be provided in the same format as returned by
+        :attr:`MDAnalysis.coordinates.base.Timestep.dimensions`:
+        ``[lx, ly, lz, alpha, beta, gamma]``.
+
+    Returns
+    -------
+    minimized_vectors : numpy.ndarray
+        Same shape and dtype as input.  The vectors from the input, but
+        minimized according to the size of the box.
+
+    .. versionadded:: 2.1.0
+    """
+    boxtype, box = check_box(box)
+    output = np.empty_like(vectors)
+
+    # use box which is same precision as input vectors
+    box = box.astype(vectors.dtype)
```

Comment:
`box` appears to always be cast down to `np.float32` in `check box` line 2410 
 ```python
box = np.asarray(box, dtype=np.float32, order='C')
```
there is the potential to then upcast to `np.float64` which is fine bc widening, but worth being aware of as supplying an `np.float64` box results in a narrowing cast.

Reply:
Its just a bit misleading at this call point as it looks like you respect the dtype of  `box` but the data has already been lost AFAIK? IDK how much this matters.

Reply:
Ok so reading through the code, there's a lot of this happening, it's bad but I think it's a little beyond this PR to fix it all. I've opened #3500